src/libopensc/card-atrust-acos.c

```diff
@@ -463,8 +463,7 @@ static int atrust_acos_select_file(struct sc_card *card,
 					bMatch += 2;
 		}
 
-		if ( card->cache.valid && bMatch >= 0 )
-		{
+		if (card->cache.valid && bMatch > 2) {
```

Comment:
on line 437 we assert the pathlen is 2 or 4 or 6
on line 440 we assert that if pathlen is 6, the first two bytes are 0x3F00
on line 444 we extend the 2 or 4 bytes paths to 4 or 6 bytes starting with 0x3F00
the line 459 is no-operation with your change. It should be removed if it keeps working as it should
This line change semantics as previously we were heading here also when we had a cache, but first two bytes of path did not match. OTOH I think this should never happen as the path should always start with 0x3F00 so I believe this change is ok.

Just remove the line 459 to avoid confusion. Same in the other files, where similar construct exist.